datalad_metalad/extractors/base.py

```diff
@@ -338,6 +344,8 @@ def get_state(self, dataset):
 # remove
 class BaseMetadataExtractor:
 
+    __generation__ = 'legacy-legacy'
```

Comment:
love the name :D

Reply:
oh god no I hate it^^

Reply:
do you think I like it?  I just used term found 4 lines above in the comment to describe this interface. If you have some other descriptive enough term to use -- let's use it  instead, I do not really care what it would be as long as we can relate/identify, e.g. `the-original`, `the-first-thought-of-michael`, `when-haxby-search-worked-best` or whatnot.

Reply:
The comment was an iteration of an old datalad-core comment. We could easily change the comment (in principle the extractor interface is independent of the storage version), we should probably just use integers to indicate the extractor generation. i.e. ``2``, ``3``, and ``4``.

```suggestion
    __generation__ = 1
```

Reply:
I think this was the `1` and we might have skipped 2 or 3 ;) but I do not care enough -- main point is to have different and somewhat informative ids here. So let's proceed with these integers